core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java

```diff
@@ -63,7 +63,7 @@ public class ZooStore<T> implements TStore<T> {
   private ZooReaderWriter zk;
   private String lastReserved = "";
   private Set<Long> reserved;
-  private Map<Long,Long> deferred;
+  private Map<Long,Duration> deferred;
```

Comment:
For correctness this probably needs to stay as Long

```suggestion
  private Map<Long,Long> deferred;
```

This is because the impl of duration has a long seconds and int nanoseconds internally.  Therefore the following program

```java
        long t1 = Long.MAX_VALUE - 30;
        long t2 = t1+40;

        System.out.println(t2-t1);

        Duration d1 = Duration.ofNanos(t1);
        Duration d2 = Duration.ofNanos(t2);

        System.out.println(d2.minus(d1));
```

prints

```
40
PT-5124095H-34M-33.709551576S
```

which is incorrect.  

The way System.nanoTime is intended to be used it relies on the behavior of subtracting longs that cross the max long boundary.  Java does not provide any good high level classes for dealing with nuanced behavior AFAIK.  There is also a java.time.Instant class that seems like it could be useful in this situation as the value for the map.  Having looked into it in the past, found Instance was also not good to use w/ System.nanoTime, but I can not remember the specifics.


This class can accept Duration as a method argument, it just needs to use Long internally.  I would recommend sticking w/ long and adding some comments about the dangers of using Duration.